lua/ide/components/bufferlist/component.lua

```diff
@@ -0,0 +1,247 @@
+local base = require("ide.panels.component")
+local libwin = require("ide.lib.win")
+local logger = require("ide.logger.logger")
+local icon_set = require("ide.icons").global_icon_set
+local commands = require("ide.components.bufferlist.commands")
+
+local BufferListComponent = {}
+
+local config_prototype = {
+	disabled_keymaps = false,
+	keymaps = {
+		edit = "<CR>",
+		edit_split = "s",
+		edit_vsplit = "v",
+		close = "d",
+	},
+}
+
+local function get_listed_bufs()
+	return vim.tbl_filter(function(bufnr)
+		return vim.bo[bufnr].buflisted and vim.api.nvim_buf_is_loaded(bufnr) and vim.api.nvim_buf_get_name(bufnr) ~= ""
```

Comment:
In libbuf we have this function: 

https://github.com/ldelossa/nvim-ide/blob/ab6e047f5006c9f78ce2037097fab28d80c60ac2/lua/ide/lib/buf.lua#L18


Can this be used instead of the current conditional?

Reply:
Looking at the code it's slightly different. `is_regular_buf` just checks that it has a buftype and that the buffer exists. Since we're using `vim.api.nvim_list_bufs` we don't need to check that it exists, and this conditional also checks if the buffer is loaded, listed, and has a name.

Reply:
cool. if you feel these checks could be used else where, you can also add an appropriate function to libbuf. If its more tailed just for the plug, this is fine too.

Reply:
I moved the whole function to libbuf 👍